cobigen-templates/templates-devon4j/src/main/java/com/devonfw/cobigen/templates/devon4j/utils/JavaUtil.java

```diff
@@ -490,4 +490,135 @@ public String getFirstEnumValue(String className) {
+            return null;
+        }
+    }
+
+    /**
+     * Helper method to get all fields recursively inclusive fields from super classes
+     *
+     * @param cl
+     *            class to find fields
+     * @param fields
+     *            list of fields to accumulate recursively
+     * @return list of all fields found
+     */
+    private static List<Field> getAllFields(List<Field> fields, Class<?> cl) {
+        fields.addAll(Arrays.asList(cl.getDeclaredFields()));
+
+        if (cl.getSuperclass() != null) {
+            getAllFields(fields, cl.getSuperclass());
+        }
+
+        return fields;
+    }
+
+    /**
+     * Helper method to get type of a field inclusive field from super classes
+     *
+     * @param pojoClass
+     *            {@link Class} the class object of the pojo
+     * @param fieldName
+     *            {@link String} the name of the field
+     * @return type of the field
+     */
+    private Class<?> getTypeOfField(Class<?> pojoClass, String fieldName) {
+        if (pojoClass != null) {
+            List<Field> fields = new ArrayList();
+            getAllFields(fields, pojoClass);
+
+            Optional<Field> field = fields.stream().filter(f -> f.getName().equals(fieldName)).findFirst();
+
+            if (field.isPresent()) {
+                return field.get().getType();
+            }
+        }
+        LOG.error("Could not find type of field {}", fieldName);
+        return null;
+    }
+
+    /**
+     * @param pojoClass
+     *            {@link Class} the class object of the pojo
+     * @param fieldName
+     *            {@link String} the name of the field
+     * @return true if the field is an instance of java.utils.Collections
+     */
+    public boolean isCollection2(Class<?> pojoClass, String fieldName) {
+        Class<?> type = getTypeOfField(pojoClass, fieldName);
+        return type == null ? false : Collection.class.isAssignableFrom(type);
+    }
+
+    /**
+     * @param className
+     *            {@link String} full qualified class name
+     * @param fieldName
+     *            {@link String} the name of the field
+     * @return type of the field in String
+     */
+    public String getCanonicalNameOfField(String className, String fieldName) {
+        try {
+            Class<?> entityClass = Class.forName(className);
+            Class<?> type = getTypeOfField(entityClass, fieldName);
+            if (type != null) {
+                return type.getCanonicalName();
+            }
+        } catch (ClassNotFoundException e) {
+            LOG.error("{}: Could not find {}", e.getMessage(), className);
+        }
+        return null;
+    }
+
+    /**
+     * Get the primary key and its type of a given Entity class
+     *
+     * @param className
+     *            full qualified class name
+     * @return the primary key and its type if found or null
+     */
+    public String getPrimaryKey(String className) {
```

Comment:
Add a test case for this method to src/test/java/com/devonfw/cobigen/templates/devon4j/test/utils/JavaUtilTest.java